lib/application.ex

```diff
@@ -0,0 +1,43 @@
+#
+# This file is part of Astarte.
+#
+# Copyright 2023 SECO Mind Srl
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+defmodule Astarte.DataAccess.Application do
+  use Application
+  require Logger
+  alias Astarte.DataAccess.Config
+
+  @app_version Mix.Project.config()[:version]
+
+  # See http://elixir-lang.org/docs/stable/elixir/Application.html
+  # for more information on OTP Applications
+  def start(_type, _args) do
+    Logger.info("Starting application v#{@app_version}.", tag: "astarte_data_access_start")
+
+    xandra_options =
+      Config.xandra_options!()
+      |> Keyword.put(:name, :astarte_data_access_xandra)
+      |> Keyword.put(:atom_keys, true)
```

Comment:
Atom keys are exposed to the caller. I agree that strings are more appropriate in the long run, however also CQEx exposed atoms (with keyword lists rather than maps). So I'd say we default the atom keys option to `true' in the first pass, so as to break less calls, and switch to strings later on.

Reply:
Then we're in case 1, so I'd do what is described above:
- Accept the `atom_keys` option from the outside, defaulting to `false`
- When you update Data Access in calling applications, you will explicitly pass `atom_keys: true` at first, putting a `TODO` in the application itself (and/or leaving an issue open to track the switch to string keys)
- When you switch to string keys in the caller, you just remove the option and it's going to default to `false`

Now, this opens up another aspect which I'm noticing just now: I was a little off when I said the library must become an `Application`, the right thing to do is actually creating the `Astarte.DataAccess` module as a [module based Supervisor](https://hexdocs.pm/elixir/1.12/Supervisor.html#module-module-based-supervisors) which basically does in `init` what is now being done in `start`, but with the advantage of allowing the application using the library to start it under its own supervision tree and passing it relevant options:
```elixir
children = [
  ...
  {Astarte.DataAccess, atom_keys: true}
  ...
]
```